### PR TITLE
Supervised finetuning minor changes

### PR DESCRIPTION
**Pull request by sanagno**
Small changes

- small typo when masking with padding
- better collator with comments
- started quantization

### CODE REVIEW

**Review by theblackcat102 — APPROVED** [review]
can we force label_mask to be boolean at the collator stage? such that we don't need .bool() in trainer.py line 73 ?

**Comment by theblackcat102** [comment]
but I think it's just minor so going to merge it

**Comment by sanagno** [comment]
Good point, that how I had it but it didn't work with deepspeed, I assume it casts tensors in a specific ways. I am going to fix it in the next commit.

File: model/supervised_finetuning/models/gptj.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
+# Taken from https://github.com/sleekmike/Finetune_GPT-J_6B_8-bit/blob/master/gpt-j-6b-8-bit.py
+
+import torch
```
Comment (mrcabbage972) [inline model/supervised_finetuning/models/gptj.py:3]: Maybe instead use bitsandbytes for a generic solution for 8-bit quantization? Are there any downsides to that?

Comment (sanagno) [reply model/supervised_finetuning/models/gptj.py:3]: Not really! feel free to edit

Comment (mrcabbage972) [reply model/supervised_finetuning/models/gptj.py:3]: Also I'm pretty sure I've previously seen a generic library that implements a whole pile of different adapter types, could be convenient.

Comment (mrcabbage972) [reply model/supervised_finetuning/models/gptj.py:3]: @sanagno Opened PR [here](https://github.com/LAION-AI/Open-Assistant/pull/582)
